File: C++ Main/ACTS/Assignment_Day08/Employee/Employee.cpp
#include <iostream>
#include "Employee.h"
#include <string>

using namespace std;

Employee::Employee()
{
	empNo = 0;
	name = "";
	dept = "";
	salary = 0;
}
	
Employee::Employee(int empNo, string name, string dept, double salary)
{
	this->empNo = empNo;
	this->name = name;
	this->dept = dept;
	this->salary = salary;
}

void Employee::Accept(int empNo, string name, string dept, double salary)
{
	this->empNo = empNo;
	this->name = name;
	this->dept = dept;
	this->salary = salary;
}

void Employee::Display()
{
	cout<<"empNo: "<<empNo<<endl;
	cout<<"Name: "<<name<<endl;
	cout<<"Dept: "<<dept<<endl;
	cout<<"Salary: "<<salary<<endl;
}

File: CDAC ACTS/_My Desktop/OOPs JAVA/Java OOPs/Day3-20240924T063043Z-001/Day3/EmployeeExample/src/com/cdac/acts/Employee.java
package com.cdac.acts;

public class Employee {
	private int empId;
	private String name;
	private String dept;
	private double salary;
	
	private static int idGenerator = 1000;
	
	public Employee(){
		this.empId = idGenerator++;
		this.name = "";
		this.dept = "";
		this.salary = 0.0;
	}
	
	public Employee(String name, String dept, double salary){
		this.empId = idGenerator++;
		this.name = name;
		this.dept = dept;
		this.salary = salary;
	}
	
	public void printData() {
		System.out.println("Employee[empId=" + empId 
				+ " name=" + name + " dept=" + dept
				+ " salary=" + salary + "]");
	}
	
}
